From fa0650be0664db0497d90566df1ad6fcfef5d1d5 Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Fri, 25 Sep 2026 19:49:22 +0100
Subject: [PATCH 2/3] disable apexd mount_before_data flag for aosp_current

apexd-bootstrap with mount_before_data enabled mounts data apexes and
processes staged sessions before /data, rebooting to bootloader on any
activation failure (bootstrap-apexd-failed). old-kernel vendors
(4.14/4.19/5.4 unisoc/exynos/qcom) fail in this path on android 17
where the same devices boot fine on android 16, where the flag was
declared but unimplemented.

disabling the flag at build time restores the android 16 bootstrap
behaviour: preinstalled bootstrap apexes only, activated via loop
devices after policy load.
---
 .../mount_before_data_flag_values.textproto                     | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/aconfig/cp2a/com.android.apex.flags/mount_before_data_flag_values.textproto b/aconfig/cp2a/com.android.apex.flags/mount_before_data_flag_values.textproto
index 56f7709..8d990f5 100644
--- a/aconfig/cp2a/com.android.apex.flags/mount_before_data_flag_values.textproto
+++ b/aconfig/cp2a/com.android.apex.flags/mount_before_data_flag_values.textproto
@@ -1,6 +1,6 @@
 flag_value {
   package: "com.android.apex.flags"
   name: "mount_before_data"
-  state: ENABLED
+  state: DISABLED
   permission: READ_ONLY
 }
-- 
2.55.0

